src/main/scala/beam/agentsim/agents/ridehail/RideHailAgent.scala

```diff
@@ -382,19 +455,36 @@ class RideHailAgent(
 
   override def logPrefix(): String = s"RideHailAgent $id: "
 
+  def requestParkingStall(): Unit = {
+    val rideHailAgentLocation =
+      RideHailAgentLocation(vehicle.driver.get, vehicle.id, vehicle.beamVehicleType.id, vehicle.spaceTime, geofence)
+    val destinationUtm = rideHailAgentLocation.currentLocationUTM.loc
+    val beta1 = 1
+    val beta2 = 1
+    val beta3 = 0.001
+    val commonUtilityParams: Map[String, UtilityFunctionOperation] = Map(
+      "energyPriceFactor" -> UtilityFunctionOperation("multiplier", -beta1),
+      "distanceFactor"    -> UtilityFunctionOperation("multiplier", -beta2),
+      "installedCapacity" -> UtilityFunctionOperation("multiplier", -beta3)
+    )
+    val mnl = new MultinomialLogit[ParkingZoneSearch.ParkingAlternative, String](Map.empty, commonUtilityParams)
+    val inquiry = ParkingInquiry(destinationUtm, "work", 0.0, mnl, 0.0)
```

Comment:
From https://github.com/LBNL-UCB-STI/beam/pull/1882 :


JustinPihony 6 days ago  Author Collaborator
@robfitzgerald @johanneshiry This is mostly a copy/paste from ChoosesParking. I know we had discussed the vot, but I wasn't sure how to work it so that fastcharging was preferred per @colinsheppard

  @rjf-nrel
rjf-nrel 6 days ago  Contributor
@JustinPihony we are actually modifying MNL, ParkingInquiry, and ChargingInquiry today as part of our integration of these things. it's different since an hour ago, so, maybe pull down the changes for now.

for prioritizing fast charging, perhaps we could add a term which is the "duration until full" cost, and, for RHA, set that multiplier higher, so it dominates over the other choices. i wonder what @johanneshiry thinks of that idea.

i'll be busy for some bits of the middle of today but i'll try and keep up with the discussion.

Reply:
@rjf-nrel @robfitzgerald @JustinPihony 
if the fast charging preference is still under discussion I like the idea of just adding such a term. I'm not sure if a "duration until full" would fit exactly what @JustinPihony is looking for or if sth. like a installed capacity multiplier (positive) would fit better, but we could both give a chance and just test it